modules/standard/Random.chpl

```diff
@@ -121,11 +121,22 @@ module Random {
     return d.isRectangular() && d.rank == 1;
 
 
-  private proc oddTimeSeed(): int(64) {
-    use Time;
-    const seed = (timeSinceEpoch().totalSeconds()*1_000_000): int;
-    const oddseed = if seed % 2 == 0 then seed + 1 else seed;
-    return oddseed;
+  private proc randomishSeed(): int {
+    import Time, IO, CTypes;
+    extern proc chpl_task_getId(): chpl_taskID_t;
+    extern proc getpid(): CTypes.c_int;
+
+    const sWho = chpl_task_getId().hash():int,
+          sWhat = getpid():int,
+          sWhen = (Time.timeSinceEpoch().totalSeconds()*1_000_000):int,
+          sWhere = here.hash():int;
+
+    try {
+      var urand = IO.openReader("/dev/urandom");
```

Comment:
`/dev/urandom` won't exist on Mac OS X, so the error handling is important here

Reply:
Are you saying that there needs to be additional error handling other than the try-catch block?

Reply:
Good point, I've refactored it